<launch>
  <!-- global parameters -->
  <arg name="map_size_x" default="30.0"/>
  <arg name="map_size_y" default="30.0"/>
  <arg name="map_size_z" default=" 5.0"/>

  <arg name="odom_topic" default="/mavros/local_position/odom" />
  <arg name="camera_pose_topic" default="/camera/pose"/>
  <arg name="depth_topic" default="/camera/depth/image_raw"/>
  <!--<arg name="depth_topic" default="/camera/depth/image_rect_raw"/>-->

  <arg name="max_acc" default="2.5" />

  <!-- main algorithm params -->
  <include file="$(find esdf_planner)/launch/esdf.xml">

    <arg name="map_size_x_" value="$(arg map_size_x)"/>
    <arg name="map_size_y_" value="$(arg map_size_y)"/>
    <arg name="map_size_z_" value="$(arg map_size_z)"/>
    <arg name="odometry_topic" value="$(arg odom_topic)"/>

    <!-- topic of point cloud measurement, such as from LIDAR  -->
    <!-- don't set camera pose and depth, if you already set this one! -->
    <arg name="cloud_topic" value="/pcl_render_node/cloud"/>

    <!-- camera pose: transform of camera frame in the world frame -->
    <!-- depth topic: depth image, 640x480 by default -->
    <!-- don't set cloud_topic if you already set these ones! -->
    <arg name="camera_pose_topic" value="$(arg camera_pose_topic)"/>
    <arg name="depth_topic" value="$(arg depth_topic)"/>

    <!-- intrinsic params of the depth camera gazebo model 640x480 -->
    <arg name="cx" value="320.5"/>
    <arg name="cy" value="240.5"/>
    <arg name="fx" value="343.49636753580074"/>
    <arg name="fy" value="343.49636753580074"/>
    <!-- maximum velocity and acceleration the drone will reach -->
    <arg name="max_acc" value="$(arg max_acc)" />
    
  </include>

</launch>
